src/header/common/link.rs

```diff
@@ -0,0 +1,1027 @@
+    /// screen.
+    Screen,
+    /// tty.
+    Tty,
+    /// tv.
+    Tv,
+    /// projection.
+    Projection,
+    /// handheld.
+    Handheld,
+    /// print.
+    Print,
+    /// braille.
+    Braille,
+    /// aural.
+    Aural,
+    /// all.
+    All,
+    /// Unrecognized media descriptor extension.
+    Extension(String)
+}
+
+/// A Link Relation Type Enum based on:
+/// [RFC5988](https://tools.ietf.org/html/rfc5988#section-6.2.2)
+#[derive(Clone, PartialEq, Debug)]
+pub enum RelationType {
+    /// alternate.
+    Alternate,
+    /// appendix.
+    Appendix,
+    /// bookmark.
+    Bookmark,
+    /// chapter.
+    Chapter,
+    /// contents.
+    Contents,
+    /// copyright.
+    Copyright,
+    /// current.
+    Current,
+    /// describedby.
+    DescribedBy,
+    /// edit.
+    Edit,
+    /// edit-media.
+    EditMedia,
+    /// enclosure.
+    Enclosure,
+    /// first.
+    First,
+    /// glossary.
+    Glossary,
+    /// help.
+    Help,
+    /// hub.
+    Hub,
+    /// index.
+    Index,
+    /// last.
+    Last,
+    /// latest-version.
+    LatestVersion,
+    /// license.
+    License,
+    /// next.
+    Next,
+    /// next-archive.
+    NextArchive,
+    /// payment.
+    Payment,
+    /// prev.
+    Prev,
+    /// predecessor-version.
+    PredecessorVersion,
+    /// previous.
+    Previous,
+    /// prev-archive.
+    PrevArchive,
+    /// related.
+    Related,
+    /// replies.
+    Replies,
+    /// section.
+    Section,
+    /// self.
+    RelationTypeSelf,
+    /// service.
+    Service,
+    /// start.
+    Start,
+    /// stylesheet.
+    Stylesheet,
+    /// subsection.
+    Subsection,
+    /// successor-version.
+    SuccessorVersion,
+    /// up.
+    Up,
+    /// versionHistory.
+    VersionHistory,
+    /// via.
+    Via,
+    /// working-copy.
+    WorkingCopy,
+    /// working-copy-of.
+    WorkingCopyOf,
+    /// ext-rel-type.
+    ExtRelType(String)
+}
+
+////////////////////////////////////////////////////////////////////////////////
+// Struct methods
+////////////////////////////////////////////////////////////////////////////////
+
+impl Link {
+    /// Create `Link` from a `Vec<LinkValue>`.
+    pub fn new(link_values: Vec<LinkValue>) -> Link {
+        Link { values: link_values }
+    }
+}
+
+#[allow(dead_code)]
+impl LinkValue {
+    /// Create `LinkValue` from URI-Reference.
+    pub fn new<T>(uri: T) -> LinkValue
+        where T: Into<Cow<'static, str>> {
+        LinkValue {
+            link: uri.into(),
+            rel: None,
+            anchor: None,
+            rev: None,
+            href_lang: None,
+            media_desc: None,
+            title: None,
+            title_star: None,
+            media_type: None,
+            link_extension: None,
+        }
+    }
+
+    /// Get the `LinkValue`'s value.
+    pub fn link(&self) -> &str {
+        self.link.as_ref()
+    }
+
+    /// Get the `LinkValue`'s `rel` parameter(s).
+    pub fn rel(&self) -> Option<&Vec<RelationType>> {
```

Comment:
I wonder, would it be nicer to return `Option<&[RelationType]>`? Is there value in getting the `&Vec`? Probably not, but maybe. Same question for each of the getters returning `&Vec`.

If we wanted to do that, it'd be a change to `opt.as_ref().map(AsRef::as_ref)`.